internal/batches/service/service.go

```diff
@@ -29,26 +30,34 @@ type Service struct {
 	client           api.Client
 	features         batches.FeatureFlags
 	imageCache       *docker.ImageCache
+	parallelism      int
 }
 
 type Opts struct {
 	AllowUnsupported bool
 	AllowIgnored     bool
 	AllowFiles       bool
 	Client           api.Client
+	Parallelism      int
 }
 
 var (
 	ErrMalformedOnQueryOrRepository = errors.New("malformed 'on' field; missing either a repository name or a query")
 )
 
 func New(opts *Opts) *Service {
+	parallelism := opts.Parallelism
+	if parallelism < 1 {
+		parallelism = 1
+	}
+
 	return &Service{
 		allowUnsupported: opts.AllowUnsupported,
 		allowIgnored:     opts.AllowIgnored,
 		allowFiles:       opts.AllowFiles,
 		client:           opts.Client,
 		imageCache:       docker.NewImageCache(),
+		parallelism:      parallelism,
```

Comment:
this sorta implies to me that service always needs to know this value, but it's only required for `EnsureDockerImages` - any low effort change we can make to only pass it into there?

Reply:
I agree with this. The `Service` and `service.Opts` structs already have a lot of fields and nobody knows how they all relate to each other.

Can't we make `parallelism int` an argument of `EnsureDockerImages`?